File: src/main/java/com/optica/repository/login/RoleRepository.java
package com.optica.repository.login;

import java.util.List;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

import com.optica.entity.Role;


@Repository
public interface RoleRepository extends JpaRepository<Role, Long> {

	List<Role> findAll();

	List<Role> findByName(String name);


}

File: src/main/kotlin/com/optica/repository/login/RoleRepository.kt
package com.optica.repository.login

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository

import com.optica.entity.Role


@Repository
interface RoleRepository : JpaRepository<Role, Long> {

    override fun findAll(): List<Role>

    fun findByName(name: String): List<Role>


}
